### PR TITLE
feat: Developer Guide - GUI - MySQL Workbench

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-gui-mysql-workbench.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+## Next steps
+
+- Learn more usage of MySQL Workbench from [the documentation of MySQL Workbench](https://dev.mysql.com/doc/workbench/en/).
+- Learn the best practices for TiDB application development with the chapters in the [Developer guide](/develop/dev-guide-overview.md), such as [Insert data](/develop/dev-guide-insert-data.md), [Update data](/develop/dev-guide-update-data.md), [Delete data](/develop/dev-guide-delete-data.md), [Single table reading](/develop/dev-guide-get-data-from-single-table.md), [Transactions](/develop/dev-guide-transaction-overview.md), and [SQL performance optimization](/develop/dev-guide-optimize-sql-overview.md).
+- Learn through the professional [TiDB developer courses](https://www.pingcap.com/education/) and earn [TiDB certifications](https://www.pingcap.com/education/certification/) after passing the exam.
+
+## Need help?
+
+Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), or [create a support ticket](https://support.pingcap.com/).
```
Comment (dveeden) [inline develop/dev-guide-gui-mysql-workbench.md:152]:
```suggestion
Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap) or [create a support ticket](https://support.pingcap.com/).
```

**Review by dveeden — APPROVED** [review]
1. Note that TiDB v7.4.0 and newer identify as MySQL 8.0.x. For workbench this means that syntax highlighting etc will work for newer syntax like `WITH ...` (CTEs).
2. I would recommend to set `Use SSL` on the `SSL` tab at least to `Require`, but maybe even more than that (e.g. `Require an Verify Identity`). This is especially true when using TiDB Cloud over an untrusted connection to the public endpoint.

File: develop/dev-guide-gui-mysql-workbench.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)
+
+6. In the setup panel, configure the following connection parameters:
+
+    - **Connection Name**: give this connection a meaningful name.
+    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
+    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
+    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
+    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Serverless cluster, then click **OK** to store the password.
```
Comment (zhangyangyu) [inline develop/dev-guide-gui-mysql-workbench.md:81]: It Keychain available on all platforms, including Windows?

Comment (dveeden) [reply develop/dev-guide-gui-mysql-workbench.md:81]: It is: https://dev.mysql.com/doc/workbench/en/wb-mysql-connections-vault.html

File: develop/dev-guide-gui-mysql-workbench.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+## Next steps
+
+- Learn more usage of MySQL Workbench from [the documentation of MySQL Workbench](https://dev.mysql.com/doc/workbench/en/).
+- Learn the best practices for TiDB application development with the chapters in the [Developer guide](/develop/dev-guide-overview.md), such as [Insert data](/develop/dev-guide-insert-data.md), [Update data](/develop/dev-guide-update-data.md), [Delete data](/develop/dev-guide-delete-data.md), [Single table reading](/develop/dev-guide-get-data-from-single-table.md), [Transactions](/develop/dev-guide-transaction-overview.md), and [SQL performance optimization](/develop/dev-guide-optimize-sql-overview.md).
+- Learn through the professional [TiDB developer courses](https://www.pingcap.com/education/) and earn [TiDB certifications](https://www.pingcap.com/education/certification/) after passing the exam.
+
+## Need help?
+
+Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap) or [create a support ticket](https://support.pingcap.com/).
```
Comment (dveeden) [inline develop/dev-guide-gui-mysql-workbench.md:152]:
```suggestion
Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap) or [create a support ticket](/support.md).
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    - **Connection Name**: give this connection a meaningful name.
+    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
+    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
+    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
+    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Dedicated cluster, then click **OK** to store the password.
+
+        ![MySQL Workbench: store Dedicated Tier password in keychain](/media/develop/mysql-workbench-store-dedicated-password-in-keychain.png)
+
+    ![MySQL Workbench: configure connection parameters panel for TiDB Dedicated](/media/develop/mysql-workbench-connection-config-dedicated-parameters.png)
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:114]:
```suggestion
   The following figure shows an example of the connection parameters:

    ![MySQL Workbench: configure connection settings for TiDB Dedicated](/media/develop/mysql-workbench-connection-config-dedicated-parameters.png)
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+6. In the setup panel, configure the following connection parameters:
+
+    - **Connection Name**: give this connection a meaningful name.
+    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
+    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
+    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
+    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Serverless cluster, then click **OK** to store the password.
+
+        ![MySQL Workbench: store password in keychain](/media/develop/mysql-workbench-store-password-in-keychain.png)
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:83]:
```suggestion
        ![MySQL Workbench: store the password of TiDB Serverless in keychain](/media/develop/mysql-workbench-store-password-in-keychain.png)
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+6. Click **Test Connection**, if the connection test is successful, you can see the **Successfully made the MySQL connection** message. Click **OK** to save the connection configuration.
+
+</div>
+<div label="TiDB Self-Hosted">
+
+1. Launch MySQL Workbench and click the **+** in the right side of **MySQL Connections**.
+
+    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)
+
+2. In the setup panel, configure the following connection parameters:
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:125]:
```suggestion
2. In the **Setup New Connection** dialog, configure the following connection parameters:
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+    - **Connection Name**: give this connection a meaningful name.
+    - **Hostname**: enter the IP address or domain name of your TiDB Self-Hosted cluster.
+    - **Port**: enter the port number of your TiDB Self-Hosted cluster.
+    - **Username**: enter the username to use to connect to your TiDB.
+    - **Password**: click **Store in Keychain ...**, enter the password to use to connect to your TiDB, then click **OK** to store the password.
+
+        ![MySQL Workbench: store self-hosted password in keychain](/media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png)
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:133]:
```suggestion
        ![MySQL Workbench: store the password of TiDB Self-Hosted in keychain](/media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png)
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+    ![MySQL Workbench: configure connection parameters panel for self-hosted TiDB](/media/develop/mysql-workbench-connection-config-self-hosted-parameters.png)
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:135]:
```suggestion

   The following figure shows an example of the connection parameters:

    ![MySQL Workbench: configure connection settings for TiDB Self-Hosted](/media/develop/mysql-workbench-connection-config-self-hosted-parameters.png)
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+3. **Test Connection** to validate the connection to the self-hosted TiDB.
+
+4. If the connection test is successful, you can see the **Connection Successful** message. Click **Save** to save the connection configuration.
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:139]:
```suggestion
3. Click **Test Connection** to validate the connection to the TiDB Self-Hosted cluster.

4. If the connection test is successful, you can see the **Successfully made the MySQL connection** message. Click **OK** to save the connection configuration.
```

File: develop/dev-guide-gui-mysql-workbench.md (hunk 3 above)
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:152]:
```suggestion
Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap), or [create a support ticket](/support.md).
```

File: media/develop/mysql-workbench-connection-config-dedicated-parameters.png
Comment (Oreoxmt) [inline media/develop/mysql-workbench-connection-config-dedicated-parameters.png:1]: Dedicated Tier is renamed to TiDB Dedicated. Please update the Connection Name to **TiDB Dedicated**.

File: media/develop/mysql-workbench-connection-config-serverless-parameters.png
Comment (Oreoxmt) [inline media/develop/mysql-workbench-connection-config-serverless-parameters.png:1]: Serverless Tier is renamed to TiDB Serverless. Please update the Connection Name to TiDB Serverless.

File: media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png
Comment (Oreoxmt) [inline media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png:1]: Please ensure that the dialog in this image maintains the same width as in media/develop/mysql-workbench-store-password-in-keychain.png and media/develop/mysql-workbench-store-dedicated-password-in-keychain.png

Comment (Icemap) [reply media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png:1]: It's not controlled by the user. And it's even unadjustable.

File: media/develop/mysql-workbench-connection-config-dedicated-parameters.png
Comment (Icemap) [reply media/develop/mysql-workbench-connection-config-dedicated-parameters.png:1]: Done

File: develop/dev-guide-gui-mysql-workbench.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+<CustomContent platform="tidb">
+
+Ask questions on the [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc), or [create a support ticket](/support.md).
+
+</CustomContent>
+
+<CustomContent platform="tidb-cloud">
+
+Ask questions on the [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc), or [create a support ticket](https://support.pingcap.com/).
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:168]:
```suggestion
Ask questions on the [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc), or [create a support ticket](/tidb-cloud/tidb-cloud-support.md).
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+2. In the **Setup New Connection** dialog, configure the following connection parameters:
+
+    - **Connection Name**: give this connection a meaningful name.
+    - **Hostname**: enter the IP address or domain name of your TiDB Self-Hosted cluster.
+    - **Port**: enter the port number of your TiDB Self-Hosted cluster.
+    - **Username**: enter the username to use to connect to your TiDB.
+    - **Password**: click **Store in Keychain ...**, enter the password to use to connect to your TiDB cluster, and then click **OK** to store the password.
+
+        ![MySQL Workbench: store the password of TiDB Self-Hosted in keychain](/media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png)
+
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:139]:
```suggestion
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+
+   The following figure shows an example of the connection parameters:
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:141]:
```suggestion
    The following figure shows an example of the connection parameters:
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+
+    - **Connection Name**: give this connection a meaningful name.
+    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
+    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
+    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
+    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Dedicated cluster, and then click **OK** to store the password.
+
+        ![MySQL Workbench: store the password of TiDB Dedicated in keychain](/media/develop/mysql-workbench-store-dedicated-password-in-keychain.png)
+
+   The following figure shows an example of the connection parameters:
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:115]:
```suggestion
    The following figure shows an example of the connection parameters:
```

File: develop/dev-guide-gui-mysql-workbench.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+
+    - **Connection Name**: give this connection a meaningful name.
+    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
+    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
+    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
+    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Serverless cluster, and then click **OK** to store the password.
+
+        ![MySQL Workbench: store the password of TiDB Serverless in keychain](/media/develop/mysql-workbench-store-password-in-keychain.png)
+
+   The following figure shows an example of the connection parameters:
```
Comment (Oreoxmt) [inline develop/dev-guide-gui-mysql-workbench.md:82]:
```suggestion
    The following figure shows an example of the connection parameters:
```

**Review by Oreoxmt — APPROVED** [review]

**Review by Mini256 — APPROVED** [review]

**Comment by dveeden** [comment]
I think we should add this to https://github.com/pingcap/docs/blob/master/develop/dev-guide-third-party-support.md#gui

**Pull request merged**